File: crates/enforcer-memory/tests/fixtures/memory/lang_cpp/widget.cpp
#include <string>
#include "widget.h"

namespace widgets {

class Widget {
public:
    Widget();
    std::string draw() const;
private:
    std::string name_;
};

std::string Widget::draw() const {
    return name_;
}

class Drawable {
public:
    virtual std::string draw() const = 0;
    virtual ~Drawable() = default;
};

class Base {
public:
    int id;
};

class DerivedWidget : public Base {
public:
    void render() {}
};

void helper_fn() {}

}  // namespace widgets

File: crates/enforcer-memory/tests/fixtures/memory/lang_dart/widget.dart
import 'dart:async';
import 'package:foo/bar.dart' as bar;

class Base {
  int id = 0;
}

abstract class Drawable {
  String draw();
}

class Widget extends Base implements Drawable {
  String name = "widget";

  @override
  String draw() {
    if (name.isEmpty) {
      return "x";
    }
    for (var i = 0; i < 3; i++) {
      helper();
    }
    return bar.render(name);
  }
}

enum Status { active, inactive }

typedef IntCallback = void Function(int x);

void helper() {}

void topLevel() {
  var w = Widget();
  w.draw();
}
